exercises/practice/roman-numerals/.approaches/itertools-starmap/content.md

```diff
@@ -44,9 +44,43 @@ def roman(number: int) -> str:
 
 ## Analysis
 
-TODO - well, it would be, right?
+The central concept is that Roman letters are defined for 1, 5 and 10, times various powers of 10.
 
+`orders` is relatively straightforward: a list of tuples, with each tuple containing the powers of 10 and (as far as possible) the letters for that number times (1, 5, 10). 
+Roman numerals for 5,000 and 10,000 are not defined, so are replaced here by spaces.
 
+The `options()` function just takes the three letters from one of these tuples and returns a list of numerals that can be constructed from them.
+For example, the 10 to 90 range:
+
+```python
+options('X', 'L', 'C')
+# => ['', 'I', 'II', 'III', 'IV', 'V', 'VI', 'VII', 'VIII', 'IX']
+```
+
+There is no zero, so that is replaced by an empty string.
+
+The `compute()` function takes a tuple from `orders` plus the top-level parameter `number`, and converts the appropriate decimal digit to its Roman equivalent, returning an `iterator`.
+For example the first digit of 723:
+
+```python
+number = 723
+[x for x in compute(100, "CDM")]
+=> ['D', 'C', 'C']
+```
+
+The `starmap()` function ties `orders`, `options()` and `compute` together, splitting up strings and tuples as necessary to give each function the parameters it needs.
+Again, an iterator is returned:
+
+```python
+number = 723
+[x for x in starmap(compute, orders)]
+# => ['', 'DCC', 'XX', 'III']
+```
+
+Finally, `''.join()` converts this iterator to a single string that can be returned as the desired answer.
+
+All clear?
+Once we get past the deliberate obfuscation, it is quite an elegant approach. Though perhaps not the most idiomatic Python!
```

Comment:
```suggestion
Once we get past the deliberate obfuscation, it is quite an elegant approach. 
Though perhaps not the most idiomatic Python.
```

Reply:
One line per sentence.  But also felt that the ! was a bit ... accusatory?  So left it out.